ils_middleware/tasks/symphony/overlay.py

```diff
@@ -10,29 +10,46 @@
 logger = logging.getLogger(__name__)
 
 
-def overlay_marc_in_symphony(*args, **kwargs) -> str:
+def overlay_marc_in_symphony(*args, **kwargs):
     """Overlays an existing record in Symphony"""
-    marc_json = kwargs.get("marc_json", "")
-    catkey = kwargs.get("catkey")
-
-    if not catkey:
-        msg = "Catalog ID is required"
-        logger.error(msg)
-        raise ValueError(msg)
-
-    marc_json = ast.literal_eval(marc_json)
-
-    payload = {
-        "@resource": "/catalog/bib",
-        "@key": catkey,
-        "catalogDate": datetime.datetime.now().strftime("%Y-%m-%d"),
-        "bib": marc_json,
-    }
-
-    return SymphonyRequest(
-        **kwargs,
-        data=json.dumps(payload),
-        http_verb="put",
-        endpoint=f"catalog/bib/key/{catkey}",
-        filter=lambda response: response.json().get("@key"),
+    task_instance = kwargs.get("task_instance")
+    resources = ast.literal_eval(
+        task_instance.xcom_pull(
+            key="overlay_resources", task_ids="process_symphony.new-or-overlay"
+        )
     )
+
+    missing_catkeys = []
+    for resource in resources:
+        resource_uri = resource["resource_uri"]
+        catkey = resource["data"].get("catkey")
+        marc_json = ast.literal_eval(
+            task_instance.xcom_pull(
+                key=resource_uri, task_ids="process_symphony.convert_to_symphony_json"
+            )
+        )
+
```

Comment:
Because we using the value string value being passed through the `kwargs`, we needed to parse the string into Python objects. The `ast.literal_eval` calls on lines **15** and **26** are causing errors and can be removed.